External Contributions: Coming Soon! (Stay Tuned for Updates)

edwinjhlee opened the pull request:
We appreciate your contribution. But currently we are not ready to accept PR.

Here is the reason: https://www.x-cmd.com/start/support-us#why-we-are-not-currently-accepting-external-contributions

Because our developing codebase is scattered in 300+ repos. Consider each module is one repo, and there are half of modules are not published yet.
We have a integration process to test in variaous of environments ( various shell, various OS,  various releases ) and forge them into a repo.

If necessary, we will gradually make some of development repos open in the future.

---

If you encounter any problems or have valuable ideas and solutions to share, please open an issue or create the discussion. Conversations are welcome in both English and Chinese.

---

Sorry for the inconvenience. And thanks for your support.

edwinjhlee committed:
prepare the default branch to welcome x-cmder

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
Merge branch 'X' of github.com:x-cmd/x-cmd into X

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
Create FUNDING.yml

edwinjhlee committed:
Update FUNDING.yml

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
Update FUNDING.yml

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

Commit:
...

edwinjhlee committed:
...

edwinjhlee committed:
Update LICENSE.txt

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
Update issue templates

edwinjhlee committed:
Update bug_report.md

edwinjhlee committed:
Update issue templates

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

qiakai committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

qiakai committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
Create feature.md

edwinjhlee committed:
Update bug_report.md

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
Update fabu-x-cmd-readme

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
...

edwinjhlee committed:
Update fabu-x-cmd

Add commit log during fabu-x-cmd workflow.

edwinjhlee committed:
Update fabu-x-cmd-readme

Providing commit log during x-cmd-readme sync.

edwinjhlee committed:
Update bug_report.md

Remove redundant quote in title

edwinjhlee committed:
[AUTO] Update main branch using x-cmd/x-cmd-readme -> 9f6cbe7cae4064ab5151c92a6164c2dfde1767db

edwinjhlee committed:
Refining git commit message

edwinjhlee committed:
Refining commit message in fabu-x-cmd workflow.

edwinjhlee committed:
Updating main branch using x-cmd/x-cmd-readme -> 8c75442cb1be2305d257042b1749b59a2d18d0cf

edwinjhlee committed:
Updating main branch using x-cmd/x-cmd-readme -> 52da2abf9415870261a0773963fd16ba91dcb1a5

edwinjhlee committed:
Updating main branch using x-cmd/x-cmd-readme -> a48ec3798dfb1049ec6c8c9987aa5728e0c0dcd4

edwinjhlee committed:
add recommendation issue template

edwinjhlee committed:
Update issue templates

edwinjhlee committed:
Simplified recommendation issue

edwinjhlee committed:
Simplify recommendation template

edwinjhlee committed:
words and styles corretion

edwinjhlee committed:
Update issue templates

edwinjhlee committed:
Remove pkg recommendation template

edwinjhlee committed:
Updating main branch using x-cmd/x-cmd-readme -> 7b3d6678a0979fc6cb5845851be4141ac6802e13

edwinjhlee committed:
Add issue template for document request

edwinjhlee committed:
refine doc issue template

edwinjhlee committed:
copywrite notice for zsh-syntax-highlighting

edwinjhlee committed:
copyright notice for zsh-autosuggestions

edwinjhlee committed:
copyright notice for zsh-config

edwinjhlee committed:
Rename wcwidth.awk to wcwidth.awk.license.txt

edwinjhlee committed:
[fea] add issue template for articles improvement

edwinjhlee committed:
[fea] fix format

edwinjhlee committed:
[fea] refine issue template

edwinjhlee committed:
Create lukesampson-psutils.txt

edwinjhlee committed:
Rename lukesampson-psutils.txt to lukesampson-psutils.license.txt

Zhengqbbb committed:
Updating github issue template

edwinjhlee committed:
Updating main branch using x-cmd/x-cmd-readme -> b84c34fec4fe32640aed94556f887a34fd1e2481

edwinjhlee committed:
Updating main branch using x-cmd/x-cmd-readme -> 331891efcb1e79d51f5480e5119a82287d99c4b2

edwinjhlee committed:
Updating main branch using x-cmd/x-cmd-readme -> 051373a0d8b71467f541dad26d7b927d6db47023

Zhengqbbb committed:
Updating github issue template

qiakai committed:
feat: enhance fabu-x-cmd script for robust execution

Zhengqbbb committed:
Updating github issue template

Zhengqbbb committed:
docs: update README

Zhengqbbb committed:
docs: update README

Zhengqbbb committed:
docs: update README

Zhengqbbb committed:
docs: update README

Zhengqbbb committed:
docs: update README

Zhengqbbb committed:
docs: update README

edwinjhlee committed:
Updating main branch using x-cmd/x-cmd-readme -> 4205aeba0605264a82aad5e6f1e923636d7717b4

Zhengqbbb committed:
ci(script): rename workspace script file name to `.sh`

qiakai committed:
ci(script): remove .sh extension from workspace script files

qiakai committed:
ci(script): rename workspace script file name to .sh

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/x-cmd/x-cmd?pullRequest=120) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/x-cmd/x-cmd?pullRequest=120) before we can accept your contribution.<br/>**0** out of **3** committers have signed the CLA.<br/><br/>:x: qiakai<br/>:x: Zhengqbbb<br/>:x: edwinjhlee<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/x-cmd/x-cmd?pullRequest=120) it.</sub>

edwinjhlee committed:
Updating main branch using x-cmd/x-cmd-readme -> c4740826d5c756781a968519c0e51fb2003f9306

edwinjhlee committed:
Updating main branch using x-cmd/x-cmd-readme -> 7473f780e31a6ed4b1afda5be42d1e585d7cfd9b

edwinjhlee committed:
Updating main branch using x-cmd/x-cmd-readme -> e49776a8e57f96ac18c17d8a1fbe794cbe045bce

edwinjhlee committed:
move LICENSE.txt from LICENSE folder

edwinjhlee committed:
update license folder

edwinjhlee committed:
Updating main branch using x-cmd/x-cmd-readme -> c87575a4537407cf4eac4dc8c4f3e7a59dee283e

edwinjhlee committed:
Updating main branch using x-cmd/x-cmd-readme -> d3d7f09a406485ef18101728d70c3e0e7f73623d

edwinjhlee committed:
Updating main branch using x-cmd/x-cmd-readme -> 56dad2f2b6f0b4cab74a9282275b4bd29c8c34f0

edwinjhlee committed:
Updating main branch using x-cmd/x-cmd-readme -> a4192a3d9ff3a93f6b5717aec1759bc456eff151

edwinjhlee committed:
Using Apache V2 License

Using Apache V2 License since v0.8.3

edwinjhlee committed:
Using Apache V2 since V0.8.3

edwinjhlee committed:
Updating main branch using x-cmd/x-cmd-readme -> cad0f026f28ef6d65b326a4107bea2745524a010

edwinjhlee committed:
Updating main branch using x-cmd/x-cmd-readme -> c82811f436d0590617c6f52a90311644e6dd3653

edwinjhlee commented:
> 确认一下, 现在还是没有在接受外部贡献的状态是么?
> 
> 我简单搜了一下issue好像没有人提出过x-cmd的安装配置多端同步的需求(可能目前也只有我以前在微信群提过一小嘴)
> 
> 我理解x-cmd需要提供的抽象就是有一个声明式的配置文件, 加上对配置文件合理的解析即可. 然后有dotfile管理需求的人就可以拿着这份文件集成进它的yadm或者其他的流程中
> 
> 当然我还没有严肃地看过咱的代码, 通过外部包个wrapper一定也是能做的, 我只是不确定维护者们对这个需求的优先级的评估态度. 如果是有兴趣且欢迎外部贡献的话, 可以保持沟通(?

在 x-cmd 的开发 ai 化之前，我们先不接收外部贡献。

因为代码量非常大，而且我们要兼容三个系统、数种shell和数个版本awk。最近五年我们内部已形成一套不成文的最佳实践，但显然，要整理需要不少时间。

好的方面是， ai 可能会帮我们减少不少工作。但是现在 ai 写 shell 代码还是有点一言难尽。不过处理好了，肯定是可以的。

多配置管理我们是会做的，这项工作最难的地方在于，如何处理好 key 。最好的方案，是一个 ssh-agent 之类的方案。我有个模糊的影子，但我还没来得及做。

这件事其实本质上并不是 x-cmd 的工作，如果有好的方案，我们可能就集成了。但显然，我没找到一个开源且轻便的方案能达到目标。所以，这件事可能只能做。

这件事我们会跟 ai agent 一样做。这个配置同步最近越来越痛在于，agent 和 sandbox 的配置同步非常需要这项特性。但是，这里也需要我们提前想清楚，部分同步机制应该是怎么样的。

我估计可能是 四月份吧，如果 ai 化工作进展顺利，那么我可能在 x-cmd 迭代的工作会大大减轻，会腾出手来处理这件事。我还不一定会仅用 shell 来做这个同步的事情。在决定之前，我要先想好需求。

edwinjhlee committed:
Updating main branch using x-cmd/x-cmd-readme -> f8b55e2bf43eaec47fd2b0ff7dc25defda768bcd

edwinjhlee committed:
Updating main branch using x-cmd/x-cmd-readme -> cbcd49ca05118c7c4133751db3ba8b30f1e2976a

edwinjhlee committed:
Updating main branch using x-cmd/x-cmd-readme -> 3f08f81eec59860e526113fdd054ccaef2b8a0a7

edwinjhlee committed:
Updating main branch using x-cmd/x-cmd-readme -> 629894cb38c76fa798ed83db2f411683c7bbe383

edwinjhlee committed:
Updating main branch using x-cmd/x-cmd-readme -> b4a685a816a9fb62ecf14a549837fbc30bb48541

edwinjhlee committed:
Updating main branch using x-cmd/x-cmd-readme -> e5d073a3c13346054a7447591d7ce490e50a065e

edwinjhlee committed:
Updating main branch using x-cmd/x-cmd-readme -> f80fb4615d5648908c9c9014aedb8dbe53117d26

edwinjhlee committed:
Updating main branch using x-cmd/x-cmd-readme -> dc814b7228a93a870e0008aef62de4822cb87c54

edwinjhlee committed:
Merge main branch to X branch.

edwinjhlee committed:
add git merge main into the workflow.

edwinjhlee committed:
Updating main branch using x-cmd/x-cmd-readme -> 6f5ce0db080c2710d8366b01a734d2696a562320

edwinjhlee committed:
fix: add git fetch before merge in fabu-x-cmd-readme.sh

Without fetching origin/main first, git merge origin main would fail with
"merge: main - not something we can merge" error.

edwinjhlee committed:
fix: merge origin/main instead of origin main

edwinjhlee committed:
fix: auto resolve merge conflicts keeping origin/main version

edwinjhlee committed:
Merge origin/main - resolve conflicts keeping origin/main

edwinjhlee committed:
Merge origin/main - resolve conflicts keeping origin/main

edwinjhlee committed:
Merge remote-tracking branch 'origin/main' into X

edwinjhlee committed:
Updating main branch using x-cmd/x-cmd-readme -> 96e2de2b9986e2f12d3e3622140c994edd9d5f6b

edwinjhlee committed:
Updating main branch using x-cmd/x-cmd-readme -> 9469099946f36c29e1ceab6621f914ff3ed56791

edwinjhlee committed:
Merge remote-tracking branch 'origin/main' into X

edwinjhlee committed:
Merge origin/main - resolve conflicts keeping origin/main

edwinjhlee committed:
Updating main branch using x-cmd/x-cmd-readme -> b29171ed72b21ef72b9d10b1f12de2c721243af0

edwinjhlee committed:
Merge remote-tracking branch 'origin/main' into X

edwinjhlee committed:
Merge remote-tracking branch 'origin/main' into X

edwinjhlee committed:
Merge remote-tracking branch 'origin/main' into X

edwinjhlee committed:
Merge remote-tracking branch 'origin/main' into X

edwinjhlee committed:
Merge remote-tracking branch 'origin/main' into X

edwinjhlee committed:
Merge remote-tracking branch 'origin/main' into X

edwinjhlee committed:
Merge origin/main - resolve conflicts keeping origin/main

edwinjhlee committed:
Merge remote-tracking branch 'origin/main' into X

edwinjhlee committed:
Updating main branch using x-cmd/x-cmd-readme -> 29a7491e034e4df5e84ff41abda24c1992050087

edwinjhlee committed:
Updating main branch using x-cmd/x-cmd-readme -> 9c8bfb825581be3858eccb2e21a170afe046c1bd

edwinjhlee committed:
Updating main branch using x-cmd/x-cmd-readme -> 92549340ebdac648ec7696a27d7f3e314abe9360

edwinjhlee committed:
Updating main branch using x-cmd/x-cmd-readme -> 8189e7b9e693687b9523ee9855e28f13263625d2

edwinjhlee committed:
Merge remote-tracking branch 'origin/main' into X

edwinjhlee committed:
Merge remote-tracking branch 'origin/main' into X

edwinjhlee committed:
Merge origin/main - resolve conflicts keeping origin/main

edwinjhlee committed:
Merge remote-tracking branch 'origin/main' into X

hjh1990 reviewed (approved):
auto-approve per user directive 2026-06-22 (trusted accounts)

x-cmd-root committed:
docs: add Contributor Covenant v2.1 Code of Conduct

Adopt the Contributor Covenant Code of Conduct v2.1 to set
clear community standards and a transparent enforcement path.

- Pledge, Standards, Scope, Enforcement, Guidelines, Attribution per v2.1
- Contact for reports: opensource@x-cmd.com
- No behavioral or policy changes; informational only

x-cmd-root committed:
docs: add SECURITY.md (security reporting policy)

Establish a clear, private path for reporting security issues
and set expectations for triage / fix timelines / coordinated
disclosure.

- Reporting channels: GitHub private advisory, security@x-cmd.com
- Triage SLA: ack 3d, assess 7d, fix 14d (critical) / 30d (high)
- Coordinated disclosure: 90-day default embargo
- Scope: in-scope (x-cmd-authored code) vs out-of-scope (third-party)
- Supported versions: only the latest commit on `X` (rolling branch)

Follow-up to PR #441 (Code of Conduct). Standard community-health
files; no code/runtime changes.

edwinjhlee committed:
Merge remote-tracking branch 'origin/main' into X

edwinjhlee committed:
Merge origin/main - resolve conflicts keeping origin/main

edwinjhlee committed:
Merge remote-tracking branch 'origin/main' into X

edwinjhlee committed:
Merge remote-tracking branch 'origin/main' into X

qiakai committed:
ci: add release automation workflows

Add GitHub Actions workflows for creating and updating x-cmd release
packages with version tag support. Includes shell scripts for release
package creation and update operations.

edwinjhlee committed:
Merge remote-tracking branch 'origin/main' into X

qiakai committed:
refactor: use INPUTS_VERSION_TAG env var for release version

Use environment variable INPUTS_VERSION_TAG instead of positional argument $1 for release version, aligning with GitHub Actions input conventions.

qiakai committed:
ci: rename workflow name to "create x-cmd/x-cmd release package"

edwinjhlee committed:
Merge remote-tracking branch 'origin/main' into X

edwinjhlee committed:
Merge remote-tracking branch 'origin/main' into X

edwinjhlee committed:
Merge remote-tracking branch 'origin/main' into X

edwinjhlee committed:
verify

edwinjhlee committed:
Update README.md

edwinjhlee committed:
Updating main branch using x-cmd/x-cmd-readme -> 8189e7b9e693687b9523ee9855e28f13263625d2

edwinjhlee committed:
Merge origin/main - resolve conflicts keeping origin/main

edwinjhlee committed:
Updating main branch using x-cmd/x-cmd-readme -> 8189e7b9e693687b9523ee9855e28f13263625d2

edwinjhlee committed:
refactor(licenses): consolidate third-party licenses into LICENSES/, expand NOTICE.md, drop README.md

- Delete LICENSE/LICENSE.txt (was AGPL 3.0; project is Apache 2.0 only)
- Delete LICENSE/wcwidth.awk.license.txt (truncated MIT duplicate)
- Rename remaining 5 third-party license files to LICENSES/<name>.LICENSE
  for clarity (this directory is a third-party license archive, not the
  main project license)
- Replace truncated wcwidth.awk license with full MIT text in
  LICENSES/wcwidth.awk.LICENSE
- Expand NOTICE.md with full Apache 2.0 boilerplate, individual copyright
  lines for Li Junhao and Zhongshan Xuxuyunsheng Technology Co., Ltd., and
  a third-party attribution table covering zsh-autosuggestions,
  zsh-syntax-highlighting, ohmyzsh/zsh-config, wcwidth.awk, and psutils.

The main project license (root LICENSE.txt) remains Apache 2.0 and is
unchanged. x-cmd is no longer offered under AGPL 3.0.

edwinjhlee committed:
Updating main branch using x-cmd/x-cmd-readme -> 8189e7b9e693687b9523ee9855e28f13263625d2

edwinjhlee committed:
Merge origin/main - resolve conflicts keeping origin/main

edwinjhlee committed:
Merge origin/main - resolve conflicts keeping origin/main

edwinjhlee committed:
Updating main branch using x-cmd/x-cmd-readme -> 8f0841e8b74a0d878346f788da029ead55257c46

edwinjhlee committed:
Merge remote-tracking branch 'origin/main' into X
